capstone/src/main/java/com/google/sps/data/ApkUnzipContent.java

```diff
@@ -0,0 +1,87 @@
+    return this.apkFilesListMap;
+  }
+
+  // Creates a constructor class for each APK with initialized map values to avoid null exceptions that may arise in Datastore from the absence of any category in APK
+  public ApkUnzipContent() {
+    apkPackageContentMap.put("res", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("dex", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("assets", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("lib", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("arsc", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+    apkPackageContentMap.put("misc", new ArrayList<Long>(Arrays.asList(0L, 0L)));
+
+  }
+
+
+
+  // Cumulates sizes of file types and file names to their keys in the maps
+  public void addApkDataToMapStorage(String nameOfFile, String typeOfFile, Long uncompressedSize, Long compressedSize) {
+    
+    ArrayList<Long> apkContentSizeDiff;
+
+    if((apkPackageContentMap.containsKey(typeOfFile))) {
+      apkContentSizeDiff = apkPackageContentMap.get(typeOfFile);
+      apkContentSizeDiff.set(0, apkContentSizeDiff.get(0) + uncompressedSize);
+      apkContentSizeDiff.set(1, apkContentSizeDiff.get(1) + compressedSize);
+      apkPackageContentMap.put(typeOfFile, apkContentSizeDiff);
+      
+    } else {
+      apkContentSizeDiff = new ArrayList<Long>();
+      apkContentSizeDiff.add(uncompressedSize);
+      apkContentSizeDiff.add(compressedSize);
+      apkPackageContentMap.put(typeOfFile, apkContentSizeDiff);
+    }
+
+    ArrayList<Long> apkFileSizeDiff = new ArrayList<Long>();
+    apkFileSizeDiff.add(uncompressedSize);
+    apkFileSizeDiff.add(compressedSize);
+    apkFilesListMap.put(nameOfFile, apkFileSizeDiff);
+    
+    
+  }
+
+  // Creates entity to be stored in Datastore using contents retrieved from the unzip function
+  public Entity toEntity(String userId, String nameOfApk, Long apkSizeOnDisk, Long totalApkSize, Long filesCount) {
+    Entity taskEntity = new Entity("UserFileFeature");
+    taskEntity.setProperty("UserId", userId);
+    taskEntity.setProperty("File_Name", nameOfApk);
+    taskEntity.setProperty("Res_File_Size", apkPackageContentMap.get("res"));
+    taskEntity.setProperty("Dex_File_Size", apkPackageContentMap.get("dex"));
+    taskEntity.setProperty("Lib_File_Size", apkPackageContentMap.get("lib"));
+    taskEntity.setProperty("Assets_File_Size", apkPackageContentMap.get("assets"));
+    taskEntity.setProperty("Resources_File_Size", apkPackageContentMap.get("arsc"));
+    taskEntity.setProperty("Misc_File_Size", apkPackageContentMap.get("misc"));
+    taskEntity.setProperty("Total_Apk_size", totalApkSize);
+    taskEntity.setProperty("Files_Count", filesCount);
+    taskEntity.setProperty("Apk_Disk_Size", apkSizeOnDisk);
+    taskEntity.setProperty("Apk_Lost_Size", (apkSizeOnDisk - totalApkSize));
+    taskEntity.setProperty("Timestamp", timestamp);
```

Comment:
What is the possible use of a timestamp? Do you have something special in mind for it?

Reply:
The timestamp helps to sort the apks from the recent ones in Datastore to display to the user.